### PR TITLE
[export] refactor generate tasks from diffs

### PR DESCRIPTION
**Pull request by arielshaqed**
Relevant to #534 (but not blocking it)

Extract long `makeTaskForDiff` internal function by separating tree-builder that builds the success file tasks from the actual diff tasks generation.

Hope it's easier on the eyes!

### CODE REVIEW

**Comment by arielshaqed** [comment]
@nopcoder asked for it so gets it for review, but happy for any other pairs of eyes to review (and approve... refactoring is better if it's better).

File: export/tasks_generator.go [hunk 1]
```diff
@@ -94,6 +94,138 @@ func NewDirMatchCache(pred func(path string) bool) *DirMatchCache {
...
+	}
+	(*s.finishedTask.TotalDependencies)++
+	return s.finishedTask.ID, nil
+}
+
+// GenerateTasksTo generates and appends all success tasks and the finished task to tasks,
+// returning a new tasks slice.
+func (s *SuccessTasksTreeGenerator) GenerateTasksTo(tasks []parade.TaskData) []parade.TaskData {
+	off := len(tasks)
+	tasks = tasks[:off+1+len(s.successTaskForDirectory)]
```
Comment (nopcoder) [inline export/tasks_generator.go:187]: this not break because you are out of bound?

File: export/tasks_generator.go [hunk 2]
```diff
@@ -94,6 +94,138 @@ func NewDirMatchCache(pred func(path string) bool) *DirMatchCache {
...
+	tasks[off] = s.finishedTask
+	return tasks
+}
+
+// makeDiffTaskBody fills TaskData *out with id, action and a body to make it a task to
+// perform diff.
+func makeDiffTaskBody(out *parade.TaskData, idGen taskIdGenerator, diff catalog.Difference, makeDestination func(string) string) error {
+	var (
+		data interface{}
+		err  error
```
Comment (nopcoder) [inline export/tasks_generator.go:201]: not sure this one is required

Comment (arielshaqed) [reply export/tasks_generator.go]: rid of it.

**Review by arielshaqed — COMMENTED** [review]
Thanks!
PTAL -- especially _is it better_?

**Review by nopcoder — APPROVED** [review]

**Pull request merged**